File: webapp/src/app/app.component.ts
import { Component } from '@angular/core';
import { Router } from '@angular/router';

@Component({
  selector: 'mix-app',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.scss']
})
export class AppComponent {
  navbarIsCollapsed = true;

  constructor(public router: Router) { }
}

File: webapp/src/main/scala/org/linnando/mixemulator/webapp/AppComponent.scala
package org.linnando.mixemulator.webapp

import angulate2.router.Router
import angulate2.std._

@Component(
  selector = "mix-emulator-app",
  templateUrl = "webapp/src/main/resources/app.component.html",
  styleUrls = @@@("webapp/src/main/resources/app.component.css")
)
class AppComponent(r: Router) {
  val router: Router = r
}
